File: mario-more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    // Prompt user for pyramid's height and validate input
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Build and print the pyramids
    for (int row = 0; row < height; row++)
    {
        // Print spaces for left pyramid
        for (int space = height - row - 1; space > 0; space--)
        {
            printf(" ");
        }

        // Print left pyramid
        for (int hash = 0; hash <= row; hash++)
        {
            printf("#");
        }

        // Print gap between pyramids
        printf("  ");

        // Print right pyramid
        for (int hash = 0; hash <= row; hash++)
        {
            printf("#");
        }

        // Move to the next line
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
from cs50 import get_int

# Prompt the user for the height
while True:
    height = get_int("Height: ")
    if height > 0 and height <= 8:
        break

# Generate the half-pyramids
for i in range(height):
    # Print spaces on the left side of the first half-pyramid
    for j in range(height - i - 1):
        print(" ", end="")

    # Print hashes for the first half-pyramid
    for j in range(i + 1):
        print("#", end="")

    # Print the gap between the half-pyramids
    print("  ", end="")

    # Print hashes for the second half-pyramid
    for j in range(i + 1):
        print("#", end="")

    # Move to the next line
    print()
